lib/Auth/Process/COmanageDbClient.php

```diff
@@ -438,6 +444,59 @@ private function getProfile($personId)
         return null;
     }
 
+  /**
+   * Get and construct the profile of the COPerson.
+   * The profile includes the attributes:
+   * given, sn, mail, voPersonVerifiedEmail, eduPersonScopedAffiliation, uid
+   *
+   * @param array &$state
+   * @param integer $co_person_id
+   * @return boolean
+   * @throws Exception
+   */
+  private function constructProfile(&$state, $co_person_id) {
+      $profile = $this->getProfile($co_person_id);
+      if (empty($profile)) {
+        return false;
+      }
+      foreach ($profile as $attributes) {
+        if (!empty($attributes['given'])) {
+          $state['Attributes']['givenName'] = array($attributes['given']);
+        }
+        if (!empty($attributes['family'])) {
+          $state['Attributes']['sn'] = array($attributes['family']);
+        }
+        if (!empty($attributes['mail'])) {
+          // XXX for the mail attribute always get the first
+          $state['Attributes']['mail'] = array(
+            explode(',', $attributes['mail'])[0]
```

Comment:
How does the sql query determine the order of the returned mail attributes ?

Reply:
The mail attribute will always contain the mail with the smallest row id value in the database.

Reply:
Effectively that means the oldest mail attribute value, right?